### PR TITLE
Remove QuadMesh and add orientation parameter to PlaneMesh

### LINKED ISSUE #4516: Merge PlaneMesh and QuadMesh into a single primitive mesh
### Describe the project you are working on

Various VFX

### Describe the problem or limitation you are having in your project

When doing billboard VFX, due to the billboard code which expects the quad to face the z axis, planes cannot be used.
But sometimes, vertex displacement can be desired even with billboard quads, but it's not possible currently to do it without an external mesh, because quads can't be subdivided.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add a "normal" property to the plane mesh and merge it with the quad mesh. 
Specifically:
Add a dropdown menu to choose between 3 options for the normal of the plane:
- UP
- FACING
- CUSTOM
When custom is selected, allow the user to enter a custom normal.
Mark Quad mesh as legacy.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

See above

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It can be worked around by either rotating the mesh in the shader or by making a quad in an external modeling software, it's just less convenient.

### Is there a reason why this should be core and not an add-on in the asset library?

It could be done with an addon but to me it is an inconsistency/ux problem of the editor. I believe that adding it to the editor will not bloat the editor itself, but will instead speed up workflow for VFX.
In the end this doesn't create new features but reworks existing ones to be more compact and consistent.

### PR DESCRIPTION
**Pull request by clayjohn**
Closes: https://github.com/godotengine/godot-proposals/issues/4516

Removes QuadMesh in favour of PlaneMesh. 

Adds ``orientation`` property to PlaneMesh which can be ``FACE_X``, ``FACE_Y``, or ``FACE_Z``. By default PlaneMesh uses ``FACE_Y``. A PlaneMesh with ``FACE_Z`` is equivalent to a QuadMesh. 

It would be nice if we could automatically convert QuadMeshes to PlaneMeshes either at load time or with the project converter tool. But I could not see an easy way of doing that as this is not a straight 1-to-1 rename. If anyone has any ideas, please let me know. 

CC @Qbieshay

### CODE REVIEW

File: scene/resources/primitive_meshes.cpp [hunk 1]
```diff
@@ -1053,13 +1066,22 @@ void PlaneMesh::_bind_methods() {
...
 	ClassDB::bind_method(D_METHOD("get_center_offset"), &PlaneMesh::get_center_offset);
 
+	ClassDB::bind_method(D_METHOD("set_orientation", "orientation"), &PlaneMesh::set_orientation);
+	ClassDB::bind_method(D_METHOD("get_orientation"), &PlaneMesh::get_orientation);
+
 	ADD_PROPERTY(PropertyInfo(Variant::VECTOR2, "size", PROPERTY_HINT_NONE, "suffix:m"), "set_size", "get_size");
 	ADD_PROPERTY(PropertyInfo(Variant::INT, "subdivide_width", PROPERTY_HINT_RANGE, "0,100,1,or_greater"), "set_subdivide_width", "get_subdivide_width");
 	ADD_PROPERTY(PropertyInfo(Variant::INT, "subdivide_depth", PROPERTY_HINT_RANGE, "0,100,1,or_greater"), "set_subdivide_depth", "get_subdivide_depth");
 	ADD_PROPERTY(PropertyInfo(Variant::VECTOR3, "center_offset", PROPERTY_HINT_NONE, "suffix:m"), "set_center_offset", "get_center_offset");
+	ADD_PROPERTY(PropertyInfo(Variant::INT, "orientation", PROPERTY_HINT_ENUM, "FACE_X, FACE_Y, FACE_Z"), "set_orientation", "get_orientation");
```
Comment (fire-forge) [inline scene/resources/primitive_meshes.cpp:1080]:
```suggestion
	ADD_PROPERTY(PropertyInfo(Variant::INT, "orientation", PROPERTY_HINT_ENUM, "Face X, Face Y, Face Z), "set_orientation", "get_orientation");
```

Property enums are formatted this way everywhere else.

**Comment by Calinou** [comment]
> It would be nice if we could automatically convert QuadMeshes to PlaneMeshes either at load time or with the project converter tool. But I could not see an easy way of doing that as this is not a straight 1-to-1 rename. If anyone has any ideas, please let me know.

I'd add a compatibility class handler so that existing projects still have some kind of visual representation visible. Otherwise, it'd be transformed into a plain Node.

File: scene/resources/primitive_meshes.cpp (hunk 1 above)
Comment (clayjohn) [reply scene/resources/primitive_meshes.cpp:1080]: Thanks for the feedback. Done!

**Review by QbieShay — APPROVED** [review]

**Comment by akien-mga** [comment]
You have a couple docs to fix too:
```
ERROR: PlaneMesh.xml: Unrecognized opening tag "QuadMesh" in class "PlaneMesh" description.
ERROR: PrimitiveMesh.xml: Unrecognized opening tag "QuadMesh" in class "PrimitiveMesh" description.
```

**Comment by akien-mga** [comment]
One more:
```
ERROR: PlaneMesh.xml: Unresolved constant reference "FACE_z" in class "PlaneMesh" description.
```

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**